frontend/src/components/assessments/assessment-creation/CreateAssessmentHeader.tsx

```diff
@@ -9,63 +10,92 @@ import {
   VStack,
 } from "@chakra-ui/react";
 
+import { TestRequest } from "../../../APIClients/types/TestClientTypes";
 import {
   EyeOutlineIcon,
   SaveOutlineIcon,
   TextOutlineIcon,
 } from "../../../assets/icons";
 import { getReadableDate } from "../../../utils/GeneralUtils";
 import BackButton from "../../common/BackButton";
+import PublishModal from "../assessment-status/EditStatusModals/PublishModal";
 
 interface CreateAssessementHeaderProps {
   name: string;
-  onSave: (event: React.MouseEvent<HTMLButtonElement>) => void;
+  handleSubmit: UseFormHandleSubmit<TestRequest>;
+  onPublish: (data: TestRequest) => Promise<void>;
+  onSave: (data: TestRequest) => Promise<void>;
+  onError: () => void;
+  validateForm: () => boolean;
 }
 
 const CreateAssessementHeader = ({
   name,
+  handleSubmit,
+  onPublish,
   onSave,
+  onError,
+  validateForm,
 }: CreateAssessementHeaderProps): React.ReactElement => {
+  const [showPublishModal, setShowPublishModal] = React.useState(false);
+  const confirmPublish = () => {
+    if (validateForm()) {
+      setShowPublishModal(true);
+    }
+  };
+
+  const handleSave = handleSubmit(onSave, onError);
+  const handlePublish = handleSubmit(confirmPublish, onError);
+  const onConfirmPublish = handleSubmit(onPublish, onError);
```

Comment:
```suggestion
  const handlePublish = handleSubmit(onPublish, onError);
  const onConfirmPublish = handleSubmit(onConfirmPublish, onError);
```
i think that makes more sense to me? (renaming)